meilisearch/_local_server.py

```
@@ -0,0 +1,233 @@
+
+        # Create a temporary directory for the binary
+        binary_dir = Path.home() / ".meilisearch" / "bin"
+        binary_dir.mkdir(parents=True, exist_ok=True)
+        binary_path = binary_dir / "meilisearch"
+
+        # Download if not already present
+        if not binary_path.exists():
+            print(f"Downloading Meilisearch binary from {download_url}...")
+            try:
+                with urlopen(download_url, timeout=300) as response:
+                    with open(binary_path, "wb") as f:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)